app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,10 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = [User.find(params[:user_id])]
+    @posts = @user.posts
+  end
```

Comment:
- [ ] The same issue is happening in the posts#index page as it is not rendering correctly, resulting in a nomethodfound error.

Cause: In the users controller, the `@user `variable is being saved as an` array` instead of an `object`. This is happening because when you use square brackets around the User.find query result like `@user = [User.find(params[:id]]` it returns an `object` which is being saved in the @user variable as an array, even if it only contains one element.

Solution: To fix this issue, you can either:

1. Remove the square brackets from around the @user variable, so that it is saved as an object:

```rb
@user = User.find(params[:id])
```

2. Access the @user variable using the @user[0] index, to access the only element in the array:

```rb

@user = [User.find(params[:id])]
@user = @user[0]

```

Recommendation: I recommend using the first solution, as it is more idiomatic and easier to read.